File: LeetCode/6020DivideArrayIntoEqualParts.java
class Solution {
    public boolean divideArray(int[] nums) {
        if(nums.length%2 != 0) return false;
        Arrays.sort(nums);
        for(int i = 0;i<nums.length;){
            if(nums[i] != nums[i+1]) return false;
            else {
                i=i+2;
            }
        }
        return true;
    }
}

//https://github.com/adityagi02/LeetCode-Weekly-Solutions/

File: LeetCode/6020DivideArrayIntoEqualParts.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        
        length = len(nums)
        nPairs = length // 2
        
        def countFreq(nums):
            n = len(nums)
            map = {}
            for i in range(n):
                if nums[i] in map.keys():
                    map[nums[i]] += 1
                else:
                    map[nums[i]] = 1
            return map
        
        mapper = countFreq(nums)
        
        if len(mapper) <= nPairs:
            for ele in mapper.values():
                if ele % 2 == 0:
                    res = True
                else:
                    return False
        else:
            res = False
        return res